added reset.rake

jpengst opened the pull request:
Fixes #issuenumber ; refs #issuenumber

Present tense short summary (50 characters or less)

More detailed description, if necessary. Try to be as descriptive as you can: even if you think that the PR content is obvious, it may not be obvious to others. Include tracebacks if helpful, and be sure to call out any bits of the PR that may be work-in-progress.

Description can have multiple paragraphs and you can use code examples inside:

``` ruby
class PostsController
  def index
    respond_with Post.limit(10)
  end
end
```

Changes proposed in this pull request:
* Added reset.rake
*
*

Guidance for testing, such as acceptance criteria or new user interface behaviors:
*
*
*

@samvera/hyrax-code-reviewers

jcoyne commented on lib/tasks/reset.rake:
```diff
@@ -0,0 +1,30 @@
+      ActiveFedora::Cleaner.clean!
+      Hyrax::PermissionTemplateAccess.delete_all
+      Hyrax::PermissionTemplate.delete_all
+      Delayed::Job.delete_all
```
Delayed::Job is not a dependency of Hyrax.

orangewolf replied:
good point - @jpengst I forgot to take this out of the copy I sent you. Unfortunately there isn't a good generic way to clear all jobs in the queue, so we should remove this and we'll make a separate task that depends on this one in our various projects.

jcoyne commented on lib/tasks/reset.rake:
```diff
@@ -0,0 +1,30 @@
+
+      # we need to wait till Fedora is done with its cleanup
+      # otherwise creating the admin set will fail
+      while AdminSet.exists?(AdminSet::DEFAULT_ID)
```
ActiveFedora::Cleaner is synchronous.  There shouldn't be a need for this.

orangewolf replied:
@jcoyne we haven't experienced it to be so. The prevailing theory was that it was async because we've seen multiple instances of the admin set creation failing after calling the cleaner. If you, for example, import 100 works, then run this task w/o the while, it will typically fail. If you run the task a second time it will succeed. Are there any other reasons it would be like that?

no-reply commented:
Hi @jpengst!

Thanks for taking the time to make this contribution. This looks like something contributed back from one of your Hyrax appplications(?), and as @jcoyne has pointed out there are a few generalization issues.

It might be helpful to start by squashing the existing changes into a commit with a message documenting what this task is useful for, and how you arrived at considerations like the `while/sleep` loop. I like to point folks to https://chris.beams.io/posts/git-commit/ for guidance about what kinds of things to consider in writing commit messages.

Thanks again! I expect this would be a handy task for a lot of folks, so I'm eager to get this one ready to merge.

jpengst commented:
@no-reply @orangewolf and I squashed the commits and updated message. Looks like @jcoyne has not had a chance to get back on the wait question. Do we want to leave the wait commented out in the rake task if people need it so that this can be merged?

cjcolvar commented:
@jpengst I'm not sure why your build has two failing tests.  Do you mind rebasing your PR on the latest master branch? I think this might clear up the failing tests.
